src/Combinatorics/Matroids/quantum_automorphism_groups.jl

```diff
@@ -0,0 +1,229 @@
+@doc raw"""
+    quantum_symmetric_group(n::Int)
+
+Return the ideal that defines the quantum symmetric group on `n` elements.
+It is comprised of `2*n + n^2 + 2*n*n*(n-1)` many generators.
+
+The relations are:
+
+  - row and column sum relations: `2*n` relations
+  - idempotent relations: `n^2` relations
+  - relations of type `u[i,j]*u[i,k]` and `u[j,i]*u[k,i]` for `k != j`: `2*n*n*(n-1)` relations
+
+# Example
+
+```jldoctest
+julia> S4 = quantum_symmetric_group(4);
+
+julia> length(gens(S4))
+120
+```
+"""
+function quantum_symmetric_group(n::Int)
+  A, u = free_associative_algebra(QQ, :u => (1:n, 1:n))
+
+  relations = elem_type(A)[]
+
+  #Idempotent relations
+  for i in 1:n, j in 1:n
+    new_relation = u[i,j] * u[i,j] - u[i,j]
+    push!(relations, new_relation)
+    for k in 1:n
+      if k != j
+        new_relation = u[i,j] * u[i,k]
+        push!(relations, new_relation)
+        new_relation = u[j,i] * u[k,i]
+        push!(relations, new_relation)
+      end
+    end
+  end
+
+  #row and column sum relations
+  for i in 1:n
+    push!(relations, sum(u[i,:]) - 1)
+    push!(relations, sum(u[:,i]) - 1)
+
+  end
+  return ideal(relations)
+end
+
+@doc raw"""
+    _quantum_automorphism_group_indices(M::Matroid, structure::Symbol=:bases)
+
+Return the indices of the relations that define the quantum automorphism group of a matroid for a given structure.
+
+# Examples
+
+```jldoctest
+julia> M = uniform_matroid(3,4);
+
+julia> idx = Oscar._quantum_automorphism_group_indices(M,:bases);
+
+julia> length(idx)
+1920
+```
+"""
+function _quantum_automorphism_group_indices(M::Matroid, structure::Symbol=:bases)
+  @req structure in [:bases, :circuits, :flats] "structure must be one of :bases, :circuits, :flats"
+
+  matroid_groundset(M) isa Vector{Int} ? nothing : M=Matroid(Oscar.pm_object(M))  
+
+  n = length(M)
+  grdSet = matroid_groundset(M)
+
+  rels = Vector{Tuple{Int,Int}}[]
+
+  sets  = getproperty(Oscar, structure)(M)
+  sizes = unique!(map(length, sets))
+
+  function _qAut_size()
+    setSizes = map(length, sets)
+
+    SetsCount = [0 for _ in 1:maximum(setSizes)]
+    foreach(size->SetsCount[size] += 1, setSizes)
+    for i in 1:length(SetsCount)
+      if SetsCount[i] == 0 || SetsCount[i] == n^i
+        SetsCount[i] = 0
+        continue
+      end
+      nonSetsCount = n^i - (SetsCount[i] * factorial(i))
```

Comment:
So the answer is yes, these sets could technically be multisets containing repetitions, but for the implemented axiom structures they don't contain any duplicates. This means that the check against `SetsCount[i] == n^i` is useless at best.

We count with `n^i -(SetsCount[i] * factorial(i))`, because `sets = getproperty(Oscar, structure)(M)` will only return one permutation for each set of the chosen structure. In the cited paper we argue that instead of choosing sets from the powerset of the groundset, we need to generalise to new bases/circuits/hyperplanes which are a subset of all tuples consisting of elements of the groundset. In the case of bases, for example, this means that a basis is a tuple such that it has no duplicate elements and the support defines a basis in the classical sense.

As an example: If `{1,2,3}` is a classical basis, then `(1,2,3), (2,1,3), (2,3,1), (1,3,2), (3,1,2), (3,2,1)` are bases in our world.

As for the name `sets`, do you have a better suggestion? I chose `sets` because it is the only term that fits all the different structures that come from basis sets, independent sets, etc. but yeah it might be confusing haha
  
I will tinker around a bit more.